### ISSUE TITLE
How to suppress stdout logging

### ISSUE THREAD

**Issue by danroot** [ggml-org/whisper.cpp#1448]
I am using a library that wraps whisper.cpp (https://github.com/sandrohanea/whisper.net) to write a cross-platform console application.  I would like to _not_ see whisper_init_state log messages in the console app.  Is there a flag or some other way to suppress logging?

**Comment by jmalfara** [comment]
Do you want to suppress all the logging? There are options in the whisper params to disable the print outs during transcriptions. You can find them in the code `whisper.cpp

**Comment by danroot** [comment]
Yes, I want to suppress or redirect all logging.  it looks like whisper_set_log_callback(nil) would do that?

**Comment by jmalfara** [comment]
`whisper_set_log_callback(nil)` will probably cause NPEs when `log(..)` is invoked. Probably best to either pass an empty function or add a boolean flag into `whisper_default_log`. I'm seeing some other `printf` or equivalents outside of that callback so there may be some more that needs to be flagged. 

You can test by commenting out the print in
```
static void whisper_default_log(const char * text) {
    fprintf(stderr, "%s", text);
}
```
and see what is still printed in the console

**Comment by contractorwolf** [comment]
I went in and commented out a bunch of of the WHISPER_LOG_INFO() out in the whisper.cpp file and some fprintf() in the main.cpp to eliminate all the logging noise (and then rebuilt with "make -j").  Also wrote a python app that takes advantage of the speed of the c++ version but allows me to pass a WAV file to it and get just the text back:

I was having similar trouble as you guys so I looked for a slightly different way.  This is the file I am using now:

my stt_wav.py file:

```
import subprocess
import sys

# Command and its arguments
command = "./whisper.cpp/main"
model = "-m ./whisper.cpp/models/ggml-base.en.bin"
file_path = "-f " + sys.argv[1]  # Get the first parameter as the wav file path

# Combine into a single command
full_command = f"{command} {model} {file_path}"

# Execute the command and capture the output
process = subprocess.Popen(full_command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
output, error = process.communicate()

if not bool(error.decode('utf-8').strip()): 
    # The output variable contains the command's standard output
    decoded_str = output.decode('utf-8').strip()
    processed_str = decoded_str.replace('[BLANK_AUDIO]', '').strip()
    print("Output:", processed_str)    
else: 
    # The error variable contains the command's standard error
    print("Error:", error.decode('utf-8'))
```

and I execute it like this:
>python stt_wav.py "./audio/wake_word_detected16k.wav"

**Comment by SalvagedMetal** [comment]
I have a circumvention, at least in C++,  ```whisper_log_set([](ggml_log_level, const char *, void *) {}, nullptr);```. This creates a callback function that ignores logs.
